vmdb/app/models/miq_provision/options_helper.rb

```diff
@@ -26,7 +26,7 @@ def get_source
     ems = source.ext_management_system
     raise MiqException::MiqProvisionError, "#{source.class.name} [#{source.name}] is not attached to a Management System" if ems.nil?
     raise MiqException::MiqProvisionError, "#{source.class.name} [#{source.name}] is attached to <#{ems.class.name}: #{ems.name}> that does not support Provisioning" unless MiqProvision::SUPPORTED_EMS_CLASSES.include?(ems.class.name)
-    raise MiqException::MiqProvisionError, "#{source.class.name} [#{source.name}] is attached to <#{ems.class.name}: #{ems.name}> with invalid credentials" if ems.authentication_invalid?
+    raise MiqException::MiqProvisionError, "#{source.class.name} [#{source.name}] is attached to <#{ems.class.name}: #{ems.name}> with invalid credentials" if ems.missing_credentials?
```

Comment:
When most of these calls were implemented we did not retaining the validity of the connection to a provider and simply verified that we had proper credentials to try to connect with a remote system.  If the connection failed the caller could deal with that.  For example this code may pass this check and still fail to connect to the provider in which case the task/request message would be updated.  So the code is in fact dealing with bad password, unreachable provider and any other condition that would cause us not to connect, just not on this line.

With that being said, if we are now relying on authentication_status_ok to indicate that we should be able to connect to the provider than it makes sense to change it here.